### ISSUE TITLE
ES is not starting up on BC3 for 5.3.0

### ISSUE THREAD

**Issue by bhavyarm** [elastic/elasticsearch#23222]
**Elasticsearch version**: 5.3.0 BC3

**Plugins installed**: [] None

**Description of the problem including expected versus actual behavior**: I just downloaded BC3 from staging for 5.3.0 and I can't start OSS ES. Tested on darwin and windows.

```
-16T17:19:30,143][INFO ][o.e.n.Node               ] version[5.3.0], pid[1123], build[cb36a8e/2017-02-16T21:50:52.046Z], OS[Mac OS X/10.11.6/x86_64], JVM[Oracle Corporation/Java HotSpot(TM) 64-Bit Server VM/1.8.0_111/25.111-b14]
[2017-02-16T17:19:30,155][WARN ][o.e.b.ElasticsearchUncaughtExceptionHandler] [] uncaught exception in thread [main]
org.elasticsearch.bootstrap.StartupException: java.lang.IllegalStateException: Unable to initialize modules
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:127) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Elasticsearch.execute(Elasticsearch.java:114) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.cli.EnvironmentAwareCommand.execute(EnvironmentAwareCommand.java:58) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.cli.Command.mainWithoutErrorHandling(Command.java:122) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.cli.Command.main(Command.java:88) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:91) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:84) ~[elasticsearch-5.3.0.jar:5.3.0]
Caused by: java.lang.IllegalStateException: Unable to initialize modules
	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:124) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.node.Node.<init>(Node.java:302) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.node.Node.<init>(Node.java:238) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Bootstrap$6.<init>(Bootstrap.java:242) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:360) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:123) ~[elasticsearch-5.3.0.jar:5.3.0]
	... 6 more
Caused by: java.nio.file.NoSuchFileException: /Users/bhavyarajumandya/Desktop/BC3/elasticsearch-5.3.0/modules/aggs-matrix-stats/plugin-descriptor.properties
	at sun.nio.fs.UnixException.translateToIOException(UnixException.java:86) ~[?:?]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:102) ~[?:?]
	at sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:107) ~[?:?]
	at sun.nio.fs.UnixFileSystemProvider.newByteChannel(UnixFileSystemProvider.java:214) ~[?:?]
	at java.nio.file.Files.newByteChannel(Files.java:361) ~[?:1.8.0_111]
	at java.nio.file.Files.newByteChannel(Files.java:407) ~[?:1.8.0_111]
	at java.nio.file.spi.FileSystemProvider.newInputStream(FileSystemProvider.java:384) ~[?:1.8.0_111]
	at java.nio.file.Files.newInputStream(Files.java:152) ~[?:1.8.0_111]
	at org.elasticsearch.plugins.PluginInfo.readFromProperties(PluginInfo.java:86) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.plugins.PluginsService.getModuleBundles(PluginsService.java:257) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:117) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.node.Node.<init>(Node.java:302) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.node.Node.<init>(Node.java:238) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Bootstrap$6.<init>(Bootstrap.java:242) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Bootstrap.setup(Bootstrap.java:242) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Bootstrap.init(Bootstrap.java:360) ~[elasticsearch-5.3.0.jar:5.3.0]
	at org.elasticsearch.bootstrap.Elasticsearch.init(Elasticsearch.java:123) ~[elasticsearch-5.3.0.jar:5.3.0]
	... 6 more
```

**Comment by rjernst** [comment]
This was fixed with fbf824f6a6e5481a569915353f4c7ae352e6f63f

**Comment by jasontedor** [comment]
Closed by fbf824f6a6e5481a569915353f4c7ae352e6f63f

**Comment by obnijnil** [comment]
I ran into the same error. My ES version:

> Version: 5.6.3, Build: 1a2f265/2017-10-06T20:33:39.012Z, JVM: 9.0.1

And this post saved me, by removing the .DS_Store files. https://discuss.elastic.co/t/error-at-starting-elasticsearch-startupexception-unable-to-initialize-modules/99266/1.
